### PR TITLE
Feature 6 : Create Projects Post API

### PR DESCRIPTION
**Pull request by Abhishek-Jain-1925**
Feature 6 : Create Projects Post API

Work Done are as follows -
1.Getting user input json and validating and decoding it.
2.Handler, Service and Repo Layer Operations.
3.Squirrel Query Mechanism and pgx toolkit for postgres Driver.
4.Unit Test Cases of API with Coverage 91.10%
5.Update the code by Suggested Changes.

### CODE REVIEW

File: internal/pkg/dto/education.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+		if edu.UniversityName == "" {
+			return errors.New("university name is required")
+		}
+
+		if edu.Place == "" {
+			return errors.New("place name is required")
+		}
+
+		fmt.Println("cgpa:", edu.PercentageOrCgpa)
```
Comment (OmkeshJosh) [inline internal/pkg/dto/education.go:39]: remove this

File: internal/pkg/dto/education.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	Degree           string `json:"degree"`
+	UniversityName  string `json:"university_name"`
+	Place            string `json:"place"`
+	PercentageOrCgpa string `json:"percent_or_cgpa"`
+	PassingYear      string `json:"passing_year"`
+}
+
+func (req *CreateEducationRequest) Validate() error {
+	if req.ProfileId == 0 {
+		return errors.New("profile_id is required")
```
Comment (OmkeshJosh) [inline internal/pkg/dto/education.go:23]: define error in errors.go file

File: internal/api/POST/profiles.go [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+import (
+	"context"
+	"net/http"
+
+	"github.com/joshsoftware/profile_builder_backend_go/internal/app/profile"
+	"github.com/joshsoftware/profile_builder_backend_go/internal/pkg/dto"
+	"github.com/joshsoftware/profile_builder_backend_go/internal/pkg/middleware"
+)
+
+func CreateProfileHandler(ctx context.Context, profileSvc profile.Service) func(http.ResponseWriter, *http.Request) {
```
Comment (OmkeshJosh) [inline internal/api/POST/profiles.go:12]: define function description

File: internal/api/POST/profiles.go [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+		err = req.Validate()
+		if err != nil {
+			middleware.ErrorResponse(w, http.StatusBadRequest, err)
+			return
+		}
+
+		err = profileSvc.CreateProfile(ctx, req)
+		if err != nil {
+			middleware.ErrorResponse(w, http.StatusBadGateway, err)
```
Comment (OmkeshJosh) [inline internal/api/POST/profiles.go:28]: It should be 500 (internal server error)

File: internal/api/POST/profiles.go [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+			return
+		}
+
+		middleware.SuccessResponse(w, http.StatusCreated, dto.MessageResponse{
```
Comment (OmkeshJosh) [inline internal/api/POST/profiles.go:32]: return profile id

**Comment by Michealjordab55** [comment]
Hello, I am looking for talented and professional freelancers, enter the necessary options:

    ~Data entry jobs

    ~Translation

    ~writing Jobs

    ~Typing jobs

    Payment: $1,500

    If you are interested kindly inbox me on my DM. Thank you
@everyone

**Pull request closed without merging**